src/core/textures/BaseTexture.js

```diff
@@ -236,9 +267,15 @@ class BaseTexture extends EventEmitter
         this.source = source;
 
         // Apply source if loaded. Otherwise setup appropriate loading monitors.
-        if ((this.source.complete || this.source.getContext) && this.source.width && this.source.height)
+        if ((source.src && source.complete || source.getContext) && source.width && source.height)
         {
-            this._sourceLoaded();
+            this._updateImageType();
+
+            if (this.imageType === 'svg'){
```

Comment:
there are a few placed here where the brace-style doesn't match the rest of the file. Please place braces on a newline